drivers/resolver/ad2s1210/ad2s1210.c

```diff
@@ -0,0 +1,408 @@
+ *
+ * @param dev - Device descriptor to create.
+ * @param init_param - Initialization parameters.
+ *
+ * @return Returns negative error code or 0 in case of success.
+ *         Example: -EINVAL - Bad input parameters.
+ *                  -ENOMEM - Failed to allocate memory.
+ *                  0 - No errors encountered.
+*******************************************************************************/
+int32_t ad2s1210_init(struct ad2s1210_dev **dev,
+		      struct ad2s1210_init_param *init_param)
+{
+	struct ad2s1210_dev *d;
+	int32_t ret;
+
+	d = (struct ad2s1210_dev *)no_os_calloc(1, sizeof(*d));
+	if (!d)
+		return -ENOMEM;
+
+	d->name = "AD2S1210";
+	d->mode = MODE_CONFIG;
+	d->resolution = init_param->resolution;
+
+	ret = no_os_gpio_get(&d->gpio_sample, &init_param->gpio_sample);
+	if (ret)
+		goto err_sample;
+
+	ret = no_os_gpio_get_optional(&d->gpio_a0, &init_param->gpio_a0);
+	if (ret)
+		goto err_a0;
+
+	ret = no_os_gpio_get_optional(&d->gpio_a1, &init_param->gpio_a1);
+	if (ret)
+		goto err_a1;
+
+	ret = no_os_gpio_get_optional(&d->gpio_res0, &init_param->gpio_res0);
+	if (ret)
+		goto err_res0;
+
+	ret = no_os_gpio_get_optional(&d->gpio_res1, &init_param->gpio_res1);
+	if (ret)
+		goto err_res1;
+
+	ret = no_os_gpio_direction_output(d->gpio_a0, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_gpio_direction_output(d->gpio_a1, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_gpio_direction_output(d->gpio_res0, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_gpio_direction_output(d->gpio_res1, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_gpio_direction_output(d->gpio_sample, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_spi_init(&d->spi_desc, &init_param->spi_init);
+	if (ret)
+		goto err_output;
+
+	if (d->gpio_a0 && d->gpio_a1)
+		d->have_mode_pins = true;
+
+	if (d->gpio_res0 && d->gpio_res1)
+		d->have_resolution_pins = true;
+
+	ret = ad2s1210_set_resolution(d, d->resolution);
+	if (ret)
+		goto err_spi;
+
+	*dev = d;
+	return 0;
+
+err_spi:
+	no_os_spi_remove(d->spi_desc);
+err_output:
+	no_os_gpio_remove(d->gpio_res1);
+err_res1:
+	no_os_gpio_remove(d->gpio_res0);
+err_res0:
+	no_os_gpio_remove(d->gpio_a1);
+err_a1:
+	no_os_gpio_remove(d->gpio_a0);
+err_a0:
+	no_os_gpio_remove(d->gpio_sample);
+err_sample:
+	no_os_free(d);
+	pr_err("%s initialization failed with status %ld\n", d->name, ret);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Returns the result of a single conversion.
+ *
+ * @param dev - The device structure.
+ * @param chn - Channel to use angle = 0, velocity = 1
+ * @param data - Buffer to store sampled register data
+ *
+ * @return 0 in case of success or negative error code.
+*******************************************************************************/
+int32_t ad2s1210_spi_single_conversion(struct ad2s1210_dev *dev,
+				       uint8_t chn,
```

Comment:
use an enum for available conversion types. Then you can drop 
```
if (chn > 1)
		return -EINVAL;
```

Reply:
ok